python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -253,25 +398,42 @@ def average_bytes_change_per_task(self) -> Optional[float]:
 
         return self.average_bytes_outputs_per_task - self.average_bytes_inputs_per_task
 
+    @property
+    def estimated_object_store_usage(self) -> Optional[float]:
```

Comment:
object store usage of an op is actually calculated as `obj_store_mem_pending_task_outputs + obj_store_mem_internal_outqueue + OpState.outqueue.memory_usage + sum(next_op.obj_store_mem_pending_task_inputs + next_op.obj_store_mem_internal_inqueue)`.
Currently it's calculated in ResourceManager [here](https://github.com/ray-project/ray/blob/94bbf99cd54bf5bf4de919755108087295d0f183/python/ray/data/_internal/execution/resource_manager.py#L133). because OpState isn't accessible here. 
It'd be also useful to report this to the dashboard.